shared/git/container.tsx

```diff
@@ -1,132 +1,94 @@
 import * as React from 'react'
 import Git, {Props as GitProps} from '.'
-import * as I from 'immutable'
 import * as GitGen from '../actions/git-gen'
 import * as RouteTreeGen from '../actions/route-tree-gen'
 import * as Constants from '../constants/git'
 import * as Types from '../constants/types/git'
 import * as Kb from '../common-adapters'
 import {anyWaiting} from '../constants/waiting'
 import * as Container from '../util/container'
-import {sortBy, partition} from 'lodash-es'
+import {sortBy} from 'lodash-es'
 import {memoize} from '../util/memoize'
-import {HeaderTitle, HeaderRightActions} from './nav-header/container'
+import {HeaderTitle, HeaderRightActions} from './nav-header'
 
 type OwnProps = {}
 
-const sortRepos = (git: Array<Types.GitInfo>) => sortBy(git, ['teamname', 'name'])
-
-const getRepos = memoize((git: I.Map<string, Types.GitInfo>) => {
-  if (!git) {
-    return {
-      personals: [],
-      teams: [],
-    }
-  }
-  const [personals, teams] = partition(git.valueSeq().toArray(), g => !g.teamname)
-
-  return {
-    personals: sortRepos(personals).map(g => g.id),
-    teams: sortRepos(teams).map(g => g.id),
-  }
-})
-
-const mapStateToProps = (state: Container.TypedState) => {
-  const {personals, teams} = getRepos(Constants.getIdToGit(state))
-  return {
-    loading: anyWaiting(state, Constants.loadingWaitingKey),
-    personals,
-    teams,
-  }
-}
-
-const mapDispatchToProps = (dispatch: Container.TypedDispatch) => ({
-  _loadGit: () => dispatch(GitGen.createLoadGit()),
-  clearBadges: () => dispatch(GitGen.createClearBadges()),
-  onBack: () => dispatch(RouteTreeGen.createNavigateUp()),
-  onNewPersonalRepo: () => {
-    dispatch(GitGen.createSetError({error: null}))
-    dispatch(RouteTreeGen.createNavigateAppend({path: [{props: {isTeam: false}, selected: 'gitNewRepo'}]}))
-  },
-  onNewTeamRepo: () => {
-    dispatch(GitGen.createSetError({error: null}))
-    dispatch(RouteTreeGen.createNavigateAppend({path: [{props: {isTeam: true}, selected: 'gitNewRepo'}]}))
-  },
-  onShowDelete: (id: string) => {
-    dispatch(GitGen.createSetError({error: null}))
-    dispatch(RouteTreeGen.createNavigateAppend({path: [{props: {id}, selected: 'gitDeleteRepo'}]}))
-  },
-})
-
-// keep track in the module
-let _expandedSet = I.Set()
-
-/* TODO
-  {} & {
-    _loadGit: (() => void)
-  } & Exclude<
-      React.ElementConfig<typeof Git>,
-      {
-        expandedSet: I.Set<string>
-        onToggleExpand: ((arg0: string) => void)
-      }
-    > */
+const getRepos = memoize((git: Map<string, Types.GitInfo>) =>
+  sortBy([...git.values()], ['teamname', 'name']).reduce<{personals: Array<string>; teams: Array<string>}>(
```

Comment:
🐑 Why did you get rid of partition?
```js
const [teams, personals] = partition(sortBy([...git.values()], ['teamname', 'name']), x => !!x.teamname)
return {teams, personals}
```

Reply:
It still partitions

Reply:
oh, sorry I missed that you had still call map(x => x.id). this is fine nvm